### ISSUE TITLE
Remove SwiftPM testing dependencies on release using Rocket.

### ISSUE THREAD

**Issue by sunshinejr** [Moya/Moya#1940]
Currently we use a pretty naive removal of testing dependencies for SwiftPM by setting an environment variable. We should be able to use Rocket and update our `Rakefile` script to just remove testing dependencies on each release, and keep it in the development process thanks to that.

The idea came from this PR that I saw https://github.com/RxSwiftCommunity/RxOptional/pull/83, which sounds really amazing.

cc @f-meloni I know you wanted to play with Rocket here as well, maybe you would like to try this one out?

**Comment by f-meloni** [comment]
Yes, that would allow me to play also a little bit with the Danger config as well.
At the moment I'm working on replacing Marathon on Danger-swift but as soon as I find a little bit of time I will have a look at this :)

**Comment by f-meloni** [comment]
I was having a look at this issue, but from what I see in the Package.swift there are actually no dev dependencies or tests that Rocket should remove

https://github.com/Moya/Moya/blob/master/Package.swift

**Comment by sunshinejr** [comment]
@f-meloni [see the Package.swift in the development branch :D](https://github.com/Moya/Moya/blob/development/Package.swift)

**Comment by f-meloni** [comment]
Ok lol 
Makes a lot of sense :D
Thank you 💯 
You should see my draft PR soon then :P
